### ISSUE TITLE
portable assembly and IPCClient co-exist ?

### ISSUE THREAD

**Issue by garyng2000** [Nethereum/Nethereum#171]
Is it possible to have both the portable assembly and IPCClient assembly co-exist ? I tried to build the Portable library(and manually copy the assembly and dependencies one to a project). did the same thing for the IPCClient. But when used together I cannot create a web3 instance using the IPC client per the document sample shown, because both assembly has the IClient interface definition.

**Comment by juanfranblanco** [comment]
Just add a reference to Nethereum.Web3 instead. Portable (eventually) will be faced off as netstandard is the new "standard". Let me know if you have any problems and your specific use case, which may justify having another IPC portable project.

**Comment by garyng2000** [comment]
yes, found that it is not possible. end up install from nuget through individual package(i.e. web3 and ipcclient). however, still have some problem to make the ipc works. I can call via the http client and I know that if I gave the wrong pipe name(i.e. not '\\.\pipe\geth.ipc'), it failed immediately. But with the right pipe, the call just stalled at sendrequest. the geth console however has no problem connecting via namedpipe. to make things simple I only created a very simple console app and whenever I use namepipe, it stalled. BTW, is there documentation on how to compile the solution ? I tried to download the master branch and still fail to compile the solution using the latest vs2017 (.5).
oh, for the use case, I am running a local geth node and the http endpoint is open to external via reverse proxy of IIS. But I need to restrict calls like web3.personal.*(making it not available in the http rpc endpoing at geth level) and only do it from C# via namedpipe ipc(i.e. webservice wrapper) as I can do additional logic on the server before the call.

**Comment by juanfranblanco** [comment]
What method are you using to hang? Ill have a look.
Compilation should not be a problem with the latest, only requirement you will need unity3d installed.

**Comment by juanfranblanco** [comment]
to solve your hanging issue try as the path:
```var web3 = new Web3(new IpcClient("geth.ipc"));```

**Comment by garyng2000** [comment]
ah, thank you so much. I thought it should be '\\.\pipe\geth.ipc' as that was what it is shown in the geth console and how I open namedpipe in the past. And yes, the unity3d probably is the reason why I get all those errors.
just found another problem but that is purely infrastructure related, the geth is running under a specific identity and the IIS app pool is running under a different identify thus the named pipe is not accessible in this situation(so my testing console app runs fine, both under the same login) but the IIS process give access deny error. Needs to think about how to deal with that.
On a related note that, is adding websocket support coming soon ?
Basically the usage scenario is that geth is both a client node and a wallet manager. for the client code(used by web3), I have to allow anonymous access but the wallet part can only be done via server C#. that is why I have to go through the ipc. The alternative would be go through websocket.
Need some thinking but really appreciate the quick help.  this 'issue' should be considered close as it is just my inexperience of the project, not a bug or anything.

**Comment by juanfranblanco** [comment]
I don't think it is a good idea to mix your clients, if you want to let geth manage your keys and transactions. The best will be to use a separate service and sign the transactions "offline" using send raw transaction. Mainly you achieve this already by using Account("private key").

Re: Web sockets, yes it is in the pipeline.

**Comment by garyng2000** [comment]
yes that is the ideal case but there are many restriction in real usage like a requirement of 'one server do it all' and people read the geth documentations and sample code and has the 'unlock account then sendTransaction' burnt into their brain :-) not that many people really understand the complexity of key management.

and you are correct that impersonation is what I end up doing. I just start geth as a sub-process in the IIS app and it is working fine.
